### PR TITLE
Human hats teshari fix

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Some code crutchs that will fix teshari's hat offsets! This ->

dreamseeker_JA56Gg0WZJ

## How This Contributes To The Skyrat Roleplay Experience

¯-_(ツ)_/¯ - I'm playing on downstream. So... I think players will be happy. 

## Proof of Testing

dreamseeker_CngRbvqnt8

## Changelog

:cl:
fix: teshari wear hardhats and other more correctly!
code: some crutchs for hatfix for teshari.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
So... I spent much time for it and didnt found any other way to fix that kind of trouble... 2 crutchs - the price of this merge. 
1) src.dir = dir.
2) A lot of "proc/update_worn_head" that uses when we put up hats. 
If anyone have any ideas that will do it without any crutch... let me know!

File: modular_skyrat/modules/customization/datums/dna.dm [hunk 1]
```diff
@@ -236,6 +236,9 @@ GLOBAL_LIST_EMPTY(total_uf_len_by_block)
 			language_holder = new species_holder(src)
 		update_atom_languages()
 
+		//Signal to handle situations when teshari rotates
+		if(mrace == /datum/species/teshari)
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/customization/datums/dna.dm]:
```suggestion
		if(istype(mrace, /datum/species/teshari)
```

Comment (ghost) [reply modular_skyrat/modules/customization/datums/dna.dm]: I'm not sure about this... It's blocks registration of the signal.

Comment (ghost) [reply modular_skyrat/modules/customization/datums/dna.dm]: The definition of mrace is datum/species/mrace. 
So... This commit was an mistake.

**Comment by ghost** [comment]
Also code doesn't work on an preview of the species.
It's coused by : src.dir = newdir.
Because signal sends BEFORE dir changes. -> code works like shit. 
And TG ... wait... i'm discovered, that i can rewrite setDir() for teshari...
Maybe it should help me with progress at this PR. 
I'm not sure.

Well... Nope. I can't do it...

**Comment by ghost** [comment]
I did not find any other way to get rid of these crutches without adding some specific variables.
For example: 
/mob/living/carbon/human.teshari_dir
. _.
I still don't want to add a signal that will be called after the atom's directory has changed.

So... That's all.

**Comment by ghost** [comment]
Thanks to https://github.com/Skyrat-SS13/Skyrat-tg/pull/21001 ! <3

**Pull request closed without merging**